packages/react-components/react-toolbar/src/components/ToolbarDivider/ToolbarDivider.tsx

```diff
@@ -3,12 +3,14 @@ import { useToolbarDividerStyles_unstable } from './useToolbarDividerStyles';
 import type { ToolbarDividerProps } from './ToolbarDivider.types';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
 import { renderDivider_unstable, useDivider_unstable } from '@fluentui/react-divider';
+import { useToolbarContext } from '../Toolbar/ToolbarContext';
 
 /**
  * ToolbarDivider component
  */
 export const ToolbarDivider: ForwardRefComponent<ToolbarDividerProps> = React.forwardRef((props, ref) => {
-  const state = useDivider_unstable({ vertical: true, ...props }, ref);
+  const { vertical } = useToolbarContext();
```

Comment:
this is enough of a reason for me to create `useToolbarDivider` as a hook, because you've broken our composition pattern by adding an 'unexpected' hook there

Reply:
Done https://github.com/microsoft/fluentui/pull/24940/commits/785ae4b50c64e4cc4db164e1a417428b6ddef245

Reply:
In that case the call to `useToolbarContext` should happen from within `useToolbarDivider`.

The whole point is that there are only ever the following function calls:
* useComponent
* useComponentContexts (optional)
* useComponentStyles
* renderComponent

Reply:
https://github.com/microsoft/fluentui/pull/24940/commits/6fed3619838c479dc2a30eb6cbad46ed5ec62e01